hphp/hack/src/hh_single_type_check.ml

```diff
@@ -506,6 +507,10 @@ let parse_options () =
       ( "--disallow-discarded-nullable-awaitables",
         Arg.Set disallow_discarded_nullable_awaitables,
         "Error on using discarded nullable awaitables" );
+      ( "--disable-xhp-element-mangling",
+        Arg.Set disable_xhp_element_mangling,
+        "Disable mangling of XHP elements :foo. This is necessary for classes using xhp class modifiers."
```

Comment:
I'm not sure this is a great comment. hh_single_type_check doesn't run code, so it's not immediately clear what it means to change mangling.

I suppose it means whether or not `<foo/>` maps to `class :foo` or `xhp class foo`.

How about something like this? "Disable mangling of XHP class names :foo. This enables use of `xhp class foo` in XHP expressions `<foo/>`".

Reply:
Agreed on the lack of clarity, though this doesn't enable it: that's `--enable-xhp-class-modifier`

this makes it so that `:foo:bar` is `\foo\bar`, not `xhp_foo__bar`, and bans `-` in XHP class names